From 1881d7bc63d694e3d7d51e03aaa3cf2951d8c64e Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:01:56 -0400
Subject: [infiniband] mlx4: Use ARRAY_SIZE instead of sizeof/sizeof[0]

Message-id: <e2594f3ad8d0eab90bca8d4925695503c2b41a68.1410425016.git.avadai@redhat.com>
Patchwork-id: 93749
O-Subject: [Patch RHEL7.1 V1 180/197] IB/mlx4: Use ARRAY_SIZE instead of sizeof/sizeof[0]
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Fabian Frederick <fabf@skynet.be>

Signed-off-by: Fabian Frederick <fabf@skynet.be>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit a57f23f675bf679b2be093571ae09fe2199c5e76)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/mlx4/main.c b/drivers/infiniband/hw/mlx4/main.c
index fed1f7a..88cf2c1 100644
--- a/drivers/infiniband/hw/mlx4/main.c
+++ b/drivers/infiniband/hw/mlx4/main.c
@@ -910,8 +910,7 @@ static int __mlx4_ib_default_rules_match(struct ib_qp *qp,
  const struct default_rules *pdefault_rules = default_table;
  u8 link_layer = rdma_port_get_link_layer(qp->device, flow_attr->port);
 
- for (i = 0; i < sizeof(default_table)/sizeof(default_table[0]); i++,
-      pdefault_rules++) {
+ for (i = 0; i < ARRAY_SIZE(default_table); i++, pdefault_rules++) {
   __u32 field_types[IB_FLOW_SPEC_SUPPORT_LAYERS];
   memset(&field_types, 0, sizeof(field_types));
 
@@ -965,8 +964,7 @@ static int __mlx4_ib_create_default_rules(
  int size = 0;
  int i;
 
- for (i = 0; i < sizeof(pdefault_rules->rules_create_list)/
-   sizeof(pdefault_rules->rules_create_list[0]); i++) {
+ for (i = 0; i < ARRAY_SIZE(pdefault_rules->rules_create_list); i++) {
   int ret;
   union ib_flow_spec ib_spec;
   switch (pdefault_rules->rules_create_list[i]) {
-- 
1.7.1